plugins/kubernetes/src/utils/clusterLinks/formatters/standard.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { standardFormatter } from './standard';
+
+function formatUrl(url: URL) {
+  // Note that we can't rely on 'url.href' since it will put the search before the hash
+  // and this won't be properly recognized by SPAs such as Angular in the standard dashboard.
```

Comment:
this is _really_ surprising to me since `path + # + hash + ? + search` makes no sense at all - the `#` has higher priority so that will have a hash of `"hash?search"` and no query at all. And that's fine as such; some SPAs indeed have a complex hash like that, but then I wonder how come that the search object had gained some data out of the hash in the first place? Did we parse with something naive like splitting by `?` instead of using the URL constructor? I think there's some underlying confusion in here

Reply:
I agree and I was puzzled too!
But let me show you:
```shell
node --version                                                                                                                                      <- kube-minikube ->
v14.17.6
```
then in the REPL:
```javascript
> x = new URL('https://foo.com/bar/?q=hello')
URL {
  href: 'https://foo.com/bar/?q=hello',
  origin: 'https://foo.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'foo.com',
  hostname: 'foo.com',
  port: '',
  pathname: '/bar/',
  search: '?q=hello',
  searchParams: URLSearchParams { 'q' => 'hello' },
  hash: ''
}
> x.hash='/account/profile'
'/account/profile'
> x
URL {
  href: 'https://foo.com/bar/?q=hello#/account/profile',
  origin: 'https://foo.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'foo.com',
  hostname: 'foo.com',
  port: '',
  pathname: '/bar/',
  search: '?q=hello',
  searchParams: URLSearchParams { 'q' => 'hello' },
  hash: '#/account/profile'
}
> x.href
'https://foo.com/bar/?q=hello#/account/profile'
```

but after a quick google search, I found this interesting answer in the Stackoverflow:
https://stackoverflow.com/questions/12682952/proper-url-forming-with-a-query-string-and-an-anchor-hashtag
which makes sense!

Indeed it says that 'Everything after # is client side' and the query parameter is interpreted by the server, not the client, and is thus part of the request.

Also, another answer says that it is really from the RFC itself!
> ? should come before the # as noted in RFC 3986:
> 
> relative-ref = relative-part [ "?" query ] [ "#" fragment ]
> 

Finally, while you were wondering how the query parsing would work, then it must be after the # to be ignored by the server and thus properly processed by the SPA which will parse the hash tag containing it!
This makes sense again!